### PR TITLE
Add k3s support

### PR DESCRIPTION
**Pull request by tobiscr**
**Description**

Adding support for K3S to the alpha-provision command.

**Related issue(s)**
#670

### CODE REVIEW

File: internal/k3s/k3s.go [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+
+	cmd := exec.CommandContext(ctx, "k3d", args...)
+
+	outBytes, err := cmd.CombinedOutput()
+	out := string(outBytes)
+	if err != nil {
+		if verbose {
+			fmt.Printf("Failing command:\n  k3d %s\nwith output:\n  %s\nand error:\n  %s\n", strings.Join(args, " "), string(out), err)
+		}
+		return out, fmt.Errorf("Executing the 'k3d %s' command with output '%s' and error message '%s' failed", strings.Join(args, " "), out, err)
```
Comment (clebs) [inline internal/k3s/k3s.go:32]:
```suggestion
		return out, errors.Wrapf(err, "Executing 'k3d %s' failed with output '%s'", strings.Join(args, " "), out)
```

File: internal/k3s/mock/help.txt [hunk 2]
```diff
@@ -0,0 +1,25 @@
+https://k3d.io/
```
Comment (clebs) [inline internal/k3s/mock/help.txt:1]: Do we really need the real help text when mocking?

File: internal/k3s/k3s_test.go [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+}
+
+// Place this folder at the beginning of PATH env-var to ensure this
+// mock-script will be used instead of a locally installed k3d tool.
+func setup() bool {
+	if os.Getenv("GOPATH") == "" {
+		fmt.Println("Could not inject k3s mock directory into PATH: env-var GOPATH is undefined")
+		return false
+	}
+	os.Setenv("PATH", fmt.Sprintf("%s:%s", os.Getenv("GOPATH")+string(os.PathSeparator)+mockDir, os.Getenv("PATH")))
```
Comment (clebs) [inline internal/k3s/k3s_test.go:34]:
```suggestion
	os.Setenv("PATH", fmt.Sprintf("%s:%s", filepath.Join(os.Getenv("GOPATH"), mockDir), os.Getenv("PATH")))
```

**Review by clebs — APPROVED** [review]

File: docs/gen-docs/kyma_alpha_provision_k3s.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
+---
+title: kyma alpha provision k3s
+---
+
+Provisions k8s cluster based on k3s.
```
Comment (klaudiagrz) [inline docs/gen-docs/kyma_alpha_provision_k3s.md:5]:
```suggestion
Provisions a Kubernetes cluster based on k3s.
```

**Review by clebs — DISMISSED** [review]

File: docs/gen-docs/kyma_alpha_provision_k3s.md [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+## Synopsis
+
+Use this command to provision a k3s cluster for Kyma installation.
```
Comment (klaudiagrz) [inline docs/gen-docs/kyma_alpha_provision_k3s.md:9]:
```suggestion
Use this command to provision a k3s-based Kubernetes cluster for Kyma installation.
```

File: docs/gen-docs/kyma_alpha_provision_k3s.md [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+```bash
+kyma alpha provision k3s [flags]
+```
+
+## Options
+
+```bash
+      --name string        Name of the Kyma cluster. (default "kyma")
+      --timeout duration   Maximum time in minutes which the provisioning takes place, where "0" means "infinite". (default 5m0s)
```
Comment (klaudiagrz) [inline docs/gen-docs/kyma_alpha_provision_k3s.md:19]:
```suggestion
      --timeout duration   Maximum time in minutes during which the provisioning takes place, where "0" means "infinite". (default 5m0s)
```

File: docs/gen-docs/kyma_alpha_provision.md [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+      --ci                  Enables the CI mode to run on CI/CD systems. It avoids any user interaction (e.g. no dialog prompts) and ensures that logs are formatted properly in log files (e.g. no spinners for CLI steps).
+  -h, --help                Displays help for the command.
+      --kubeconfig string   Specifies the path to the kubeconfig file. By default, Kyma CLI uses the KUBECONFIG environment variable or "/$HOME/.kube/config" if the variable is not set.
+      --non-interactive     Enables the non-interactive shell mode.
+  -v, --verbose             Displays details of actions triggered by the command.
+```
+
+## See also
+
+* [kyma alpha](#kyma-alpha-kyma-alpha)	 - Executes the commands in alpha testing stage.
```
[the remark this answers is not in the document]
Comment (tobiscr) [reply docs/gen-docs/kyma_alpha_provision.md:23]: Fixed in code

File: docs/gen-docs/kyma_alpha_provision.md [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+* [kyma alpha provision k3s](#kyma-alpha-provision-k3s-kyma-alpha-provision-k3s)	 - Provisions k8s cluster based on k3s.
```
[the remark this answers is not in the document]
Comment (tobiscr) [reply docs/gen-docs/kyma_alpha_provision.md:24]: fixed in code

File: docs/gen-docs/kyma_alpha.md [hunk 9]
```diff
@@ -22,5 +22,6 @@ Use this command to utilize the Kyma CLI in alpha testing stage.
 
 * [kyma](#kyma-kyma)	 - Controls a Kyma cluster.
 * [kyma alpha deploy](#kyma-alpha-deploy-kyma-alpha-deploy)	 - Deploys Kyma on a running Kubernetes cluster.
+* [kyma alpha provision](#kyma-alpha-provision-kyma-alpha-provision)	 - Provisions a cluster for Kyma installation.
```
[the remark this answers is not in the document]
Comment (tobiscr) [reply docs/gen-docs/kyma_alpha.md:25]: That's currently handled by the doc-generator :/

File: docs/gen-docs/kyma_alpha_provision_k3s.md (hunk 4 above)
Comment (tobiscr) [reply docs/gen-docs/kyma_alpha_provision_k3s.md:5]: fixed in code

File: docs/gen-docs/kyma_alpha_provision_k3s.md (hunk 5 above)
Comment (tobiscr) [reply docs/gen-docs/kyma_alpha_provision_k3s.md:9]: fixed in code

File: docs/gen-docs/kyma_alpha_provision_k3s.md (hunk 6 above)
Comment (tobiscr) [reply docs/gen-docs/kyma_alpha_provision_k3s.md:19]: fixed in code

**Review by clebs — APPROVED** [review]

**Review by klaudiagrz — APPROVED** [review]

**Pull request merged**